css: Let drag-to-resize expand settings textareas to panel width.

saubhagya-patel opened the pull request:
This PR ensures that the drag-to-resize handles for textarea inputs in organization settings allow resizing to the full panel width.

The PR completes the following remaining task of #35663:
> The drag-to-resize handles for the organization settings input don't let you make it as wide as the panel, which seems odd.

<!-- Describe your pull request here.-->

Fixes: #35663.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**


---


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
@alya @sahil839 can your review this one? We may want to backport it if we think it's safe and helpful.

alya commented:
We should also put reasonable minimum constraints on resizing these text boxes, to avoid weird situations:

<img width="256" height="110" alt="Screenshot 2025-08-29 at 12 52 28" src="https://github.com/user-attachments/assets/67a06221-4d6f-434b-be82-8b2fa2b8d4b6" />

I'd suggest the default width as the min width, and maybe 2 lines of text height or so as the default height.

saubhagya-patel commented:
I have updated the PR with the following changes:
- Limited the max-width to 80%.
- Added a default min-height and min-width.

@alya, I have updated the PR. I hope the updated changes look good to you. Please let me know if any further adjustments are needed.
Thank you.

alya commented:
The min height is not what I suggested, and I don't think it works well. I said:

> I'd suggest the default width as the min width, and maybe 2 lines of text height or so as the default height.

alya commented:
The rest looks fine to me.

saubhagya-patel commented:
I have updated the PR with the following changes:
- Added a default height of 2 lines.
- Added a `min-height` of `2em`.

@alya, I have updated the PR. I hope the updated changes look good to you. Please let me know if any further adjustments are needed.
Thank you.

alya commented:
Feels good in my manual testing!

@sahil839 could you please take a look?

sahil839 commented on web/src/settings_org.ts:
```diff
@@ -457,6 +457,21 @@ function update_test_welcome_bot_custom_message_button_status(): void {
+
+    const min_rows = 2;
+    const max_rows = 5;
+    $el.attr("rows", min_rows);
```
What is the need to set `rows` attribute here when it is being set in the end?

sahil839 replied:
Surprisingly, it works for me after removing this line. Also theoretically, it should not be required because we anyways set the height at the end of this function and `scrollHeight` or `line_height` should not depend on the `rows` attribute.
![Peek 2025-09-18 12-50](https://github.com/user-attachments/assets/12998bab-20b9-4ea6-b0ee-e41aed319d46)

sahil839 commented on web/src/settings_org.ts:
```diff
@@ -1142,6 +1157,8 @@ export function save_organization_settings(
                 // a change.
                 unsaved_welcome_message_custom_text = "";
             }
+            resize_settings_textarea("id_realm_welcome_message_custom_text");
```
Instead of calling this for changing every setting, I think the function can take subsection element as input and then return early if there is no textarea in the subsection and resize them if there is one. 

And for the function calls in `settings_panel_menu`, there can be two ways - it can just call a separate function which can call this function in loop for all the subsections in section (this will handle other textarea elements added in future without having to update the code for them) or it can just call this function for two subsections based on the `section` value.

sahil839 commented on web/templates/settings/organization_profile_admin.hbs:
```diff
@@ -35,7 +35,7 @@
                     <label for="id_realm_description" class="settings-field-label">{{t "Organization description" }}</label>
                     <textarea id="id_realm_description" name="realm_description" class="admin-realm-description setting-widget prop-element settings_textarea"
-                      maxlength="1000" data-setting-widget-type="string">{{ realm_description }}</textarea>
+                      maxlength="1000" data-setting-widget-type="string" rows="2">{{ realm_description }}</textarea>
```
https://developer.mozilla.org/en-US/docs/Web/API/HTMLTextAreaElement/rows mentions 2 is the default value, so maybe we can skip this and we anyways set this value when opening the panel.

sahil839 commented on web/src/settings_components.ts:
```diff
@@ -2017,3 +2017,42 @@ export function get_channel_folder_value_from_dropdown_widget($elem: JQuery): nu
+        const needed_rows = Math.ceil(scrollheight / line_height) - 1;
+
+        const new_rows = Math.min(Math.max(needed_rows, min_rows), max_rows);
+        $el.attr("rows", new_rows);
```
I guess `settings_org.resize_settings_textarea` can be moved to this file and then be used here directly.

sahil839 commented:
I can reproduce it by calling `$el.attr("rows", new_rows);` with a delay using `setTimeout`, though I am not sure how much will we able to reproduce that actually.
[Screencast from 2025-10-26 19-05-50.webm](https://github.com/user-attachments/assets/97731c9c-8ed5-4012-ac1d-7a0a6f8e059b)

@timabbott I think you can review this now. Looks good to me other than the potential flickering issue.
